### PR TITLE
Refactor RsFork usages to TkFork

### PR DESCRIPTION
**Pull request by Happy-Neko**
Fix for #635.

*  Refactor `TkProduces` to use `TkFork`.
*  Refactor `TkConsumes` to use `TkFork`.
*  Refactor `TkGzip` to use `TkFork`.

### CODE REVIEW

**Comment by Happy-Neko** [comment]
@davvd Please proceed with code review.

**Comment by davvd** [comment]
@happy-neko Thanks, I will ask someone to do a code review here, thanks for your pull request

**Comment by davvd** [comment]
@krzyk review this one,please

**Comment by krzyk** [comment]
@davvd please assign someone else

**Comment by davvd** [comment]
> @davvd please assign someone else

@krzyk 15 points was deducted from your rating

**Comment by davvd** [comment]
> @davvd please assign someone else

@krzyk no problem, I will try to find somebody else

**Comment by davvd** [comment]
@dmzaytsev please review, thanks

File: src/main/java/org/takes/facets/fork/FkContentType.java [hunk 1]
```diff
@@ -58,15 +56,6 @@
     /**
      * Ctor.
      * @param atype Accepted type
-     * @param response Response to return
-     */
-    public FkContentType(final String atype, final Response response) {
```
Comment (dmzaytsev) [inline src/main/java/org/takes/facets/fork/FkContentType.java]: @Happy-Neko why did you remove this constructor?
I think we should leave it as is
maybe some clients already uses this constructor

File: src/main/java/org/takes/facets/fork/FkEncoding.java [hunk 2]
```diff
@@ -71,18 +71,18 @@
...
-    private final transient Response origin;
+    private final transient Take origin;
 
     /**
      * Ctor.
      * @param enc Encoding we accept
-     * @param response Response to return
+     * @param take Take to handle the request dynamically.
      */
-    public FkEncoding(final String enc, final Response response) {
```
Comment (dmzaytsev) [inline src/main/java/org/takes/facets/fork/FkEncoding.java]: @Happy-Neko same here I think we should leave this constructor for compatibility  and create a new one like `FkContentType` 
@yegor256 what do you think?

File: src/main/java/org/takes/facets/fork/FkTypes.java [hunk 3]
```diff
@@ -49,27 +49,27 @@
...
-    private final transient Response origin;
+    private final transient Take origin;
 
     /**
      * Ctor.
      * @param list List of types
-     * @param response Response to return
+     * @param take Take to handle the request dynamically.
      */
-    public FkTypes(final String list, final Response response) {
```
Comment (dmzaytsev) [inline src/main/java/org/takes/facets/fork/FkTypes.java]: @Happy-Neko  I think we should leave this constructor for compatibility

**Comment by dmzaytsev** [comment]
@Happy-Neko why did you remove `RsFork`?
the original ticket says nothing about it

**Comment by dmzaytsev** [comment]
@Happy-Neko please see a few comments above

**Comment by dmzaytsev** [comment]
@Happy-Neko please close the pull request if you stopped working on the ticket

**Comment by davvd** [comment]
@ypshenychka please, help us to evaluate quality here, as per [par.24](http://at.teamed.io/policy.html#24)

**Comment by ypshenychka** [comment]
@davvd Quality is acceptable.

**Comment by davvd** [comment]
> @davvd Quality is acceptable.

@ypshenychka thanks for the QA review, we'll work better next time

**Comment by davvd** [comment]
@dmzaytsev I added 10 mins to @ypshenychka (for QA review) in transaction 82900447. Thanks so much! Your account was topped up for **21 mins** (transaction ID is `82900460`, the task took 425 hours and 38 mins). you're getting extra minutes here (c=6). added +21 to your rating, now it is equal to [+2854](http://www.netbout.com/b/35166?open=rating)
